### PR TITLE
Adds some comments and refactors some of the presubmit-check for clarity

### PR DESCRIPTION
**Pull request by jaboc83**
Some of the functions had missing or incorrect comments so this change adds
some more comments for consistency and makes a change to use a vinyl file
where it was specified in the comments but not used yet in the code. Also,
I renamed one of the functions to better reflect what was being tested (any vs all)
as it didn't make sense based on the current logic.

### CODE REVIEW

File: build-system/tasks/presubmit-checks.js [hunk 1]
```diff
@@ -126,11 +126,16 @@ var requiredTerms = {
...
 /**
- * @param {!File} file file is a vinyl file object
- * @param {!Array<string, string>} terms
- * @return boolean
+ * Logs any issues found in the contents of file based on terms (regex patterns),
+ * and provides any possible fix information for matched terms if possible
+ *
+ * @param {!File} file file is a vinyl file object to match terms on content with
+ * @param {!Array<string, string>} terms Pairs of regex patterns and possible fix messages.
+ * @return boolean true if any of the terms match the file content, false otherwise
```
Comment (erwinmombay) [inline build-system/tasks/presubmit-checks.js]: thanks :+1:

Comment (erwinmombay) [inline build-system/tasks/presubmit-checks.js]: oh can you change this to `@return {boolean}`

File: build-system/tasks/presubmit-checks.js [hunk 2]
```diff
@@ -158,25 +164,36 @@ function matchTerms(pathname, contents, terms) {
 }
 
 
+/**
+ * Test if a file's contents match any of the
+ * forbidden terms
+ *
+ * @param {!File} file file is a vinyl file object
+ * @return boolean true if any of the terms match the file content, false otherwise
```
Comment (erwinmombay) [inline build-system/tasks/presubmit-checks.js]: same here

File: build-system/tasks/presubmit-checks.js [hunk 3]
```diff
@@ -158,25 +164,36 @@ function matchTerms(pathname, contents, terms) {
...
 }
 
-function hasAllTerms(file) {
+/**
+ * Test if a file's contents fail to match any of the required terms and log any missing terms
+ *
+ * @param {!File} file file is a vinyl file object
+ * @return boolean true if any of the terms are not matched in the file content, false otherwise
+ */
+function isMissingTerms(file) {
```
Comment (erwinmombay) [inline build-system/tasks/presubmit-checks.js]: and here. lol these are all my mistakes. im sorry

File: build-system/tasks/presubmit-checks.js (hunk 1 above)
Comment (jaboc83) [inline build-system/tasks/presubmit-checks.js]: Sure thing!

**Comment by jaboc83** [comment]
Bah! I need a line length checker setup in my vimrc...

**Comment by erwinmombay** [comment]
@jaboc83 something like

```vim
set colorcolumn=80
highlight ColorColumn guibg=Gray14
```

should work

**Comment by jaboc83** [comment]
Ha, yeah, I just added the exact same thing

File: build-system/tasks/presubmit-checks.js [hunk 4]
```diff
@@ -126,11 +126,19 @@ var requiredTerms = {
...
 
 /**
- * @param {!File} file file is a vinyl file object
- * @param {!Array<string, string>} terms
- * @return boolean
+ * Logs any issues found in the contents of file based on terms (regex
+ * patterns), and provides any possible fix information for matched terms if
+ * possible
+ *
+ * @param {!File} file a vinyl file object to match terms on content with
```
Comment (erwinmombay) [inline build-system/tasks/presubmit-checks.js]: sentence feels incomplete, maybe remove `with` ?

**Comment by erwinmombay** [comment]
LGTM! thanks

**Pull request merged**